lib/msal-core/src/telemetry/ApiEvent.ts

```diff
@@ -0,0 +1,91 @@
+import TelemetryEvent from "./TelemetryEvent";
+import { EVENT_NAME_PREFIX, TELEMETRY_BLOB_EVENT_NAMES } from "./TelemetryConstants";
+import { scrubTenantFromUri, hashPersonalIdentifier } from "./TelemetryUtils";
+import { Logger } from "../Logger";
+
+const AUTHORITY_KEY = EVENT_NAME_PREFIX + "authority";
+const AUTHORITY_TYPE_KEY = EVENT_NAME_PREFIX + "authority_type";
+const PROMPT_KEY = EVENT_NAME_PREFIX + "ui_behavior";
+const TENANT_ID_KEY = EVENT_NAME_PREFIX + "tenant_id";
+const USER_ID_KEY = EVENT_NAME_PREFIX + "user_id";
+const WAS_SUCESSFUL_KEY = EVENT_NAME_PREFIX + "was_successful";
+const API_ERROR_CODE_KEY = EVENT_NAME_PREFIX + "api_error_code";
+const LOGIN_HINT_KEY = EVENT_NAME_PREFIX + "login_hint";
+
+export const API_CODE = {
```

Comment:
We could possibly simplify this with the changes in PR #792 so we are just tracking silent and interactive events.

Reply:
Could you expand a little more :D

Reply:
#792 changes our internal structure so we just have acquireTokenInteractive and acquireTokenSilent, while the interaction type (popup or redirect) is handled by a UI function, so we have separated duties a little bit. We can simplify our telemetry events by just having "interactive" and "silent" events and just logging whether they are getting an id_token, access_token, and using redirect or popup.

Reply:
I think it is valuable to track both, Api events will tell us what users are using